database/migrations/new/2021_03_16_095556_create_todos_table.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+class CreateTodosTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('todos', function (Blueprint $table) {
+            //
+            $table->increments('id');
+            $table->unsignedBigInteger('customer_id');
+            $table->text('todo');
+            $table->unsignedInteger('user_id');
+            $table->foreign('user_id')->references('id')->on('users');
+            $table->foreign('customer_id')->references('id')->on('contact_customers');
+            $table->string('type');
+            $table->string('status');
+            $table->date('due_date')->nullable();
+            $table->timestamps();
+        });
+    }
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::table('todos', function (Blueprint $table) {
+            $table->dropForeign('todos_customer_id_foreign');
```

Comment:
why are we dropping foreign key first???

Reply:
I encounterd foreign key error when rolling back. I had to drop foreign key first